go/runtime/host/composite/composite.go

```diff
@@ -0,0 +1,162 @@
+	return c.comps[bundle.ComponentRONL].ID()
+}
+
+// Implements host.Runtime.
+func (c *composite) GetInfo(ctx context.Context) (*protocol.RuntimeInfoResponse, error) {
+	return c.comps[bundle.ComponentRONL].GetInfo(ctx)
+}
+
+// Implements host.Runtime.
+func (c *composite) GetCapabilityTEE() (*node.CapabilityTEE, error) {
+	return c.comps[bundle.ComponentRONL].GetCapabilityTEE()
+}
+
+// shouldPropagateToComponent checks whether the given runtime request should be propagated to the
+// given component call.
+func shouldPropagateToComponent(body *protocol.Body) bool {
+	switch {
+	case body.RuntimeConsensusSyncRequest != nil:
+		// Consensus view of all components should be up to date as otherwise signed attestations
+		// will be stale, resulting in them being rejected.
+		return true
+	case body.RuntimeKeyManagerPolicyUpdateRequest != nil,
+		body.RuntimeKeyManagerStatusUpdateRequest != nil,
+		body.RuntimeKeyManagerQuotePolicyUpdateRequest != nil:
+		// Key manager updates should be propagated.
+		return true
+	default:
+		return false
+	}
+}
+
+// Implements host.Runtime.
+func (c *composite) Call(ctx context.Context, body *protocol.Body) (*protocol.Body, error) {
+	result, err := c.comps[bundle.ComponentRONL].Call(ctx, body)
+	if err != nil {
+		return nil, err
+	}
+
+	for kind, comp := range c.comps {
+		if kind == bundle.ComponentRONL {
+			continue // Already handled above.
+		}
+		if !shouldPropagateToComponent(body) {
+			continue
+		}
+
+		_, err = comp.Call(ctx, body)
+	}
+	return result, nil
+}
+
+// Implements host.Runtime.
+func (c *composite) UpdateCapabilityTEE() {
+	for _, comp := range c.comps {
+		comp.UpdateCapabilityTEE()
+	}
+}
+
+// Implements host.Runtime.
+func (c *composite) WatchEvents(ctx context.Context) (<-chan *host.Event, pubsub.ClosableSubscription, error) {
+	return c.comps[bundle.ComponentRONL].WatchEvents(ctx)
+}
+
+// Implements host.Runtime.
+func (c *composite) Start() error {
+	for kind, comp := range c.comps {
+		if err := comp.Start(); err != nil {
+			return fmt.Errorf("host/composite: failed to start component '%s': %w", kind, err)
+		}
+	}
+	return nil
+}
+
+// Implements host.Runtime.
+func (c *composite) Abort(ctx context.Context, force bool) error {
+	// Only RONL supports aborts.
+	return c.comps[bundle.ComponentRONL].Abort(ctx, force)
+}
+
+// Implements host.Runtime.
+func (c *composite) Stop() {
+	for _, comp := range c.comps {
+		comp.Stop()
+	}
+}
+
+// Implements host.CompositeRuntime.
+func (c *composite) Component(kind bundle.ComponentKind) host.Runtime {
+	return c.comps[kind]
+}
+
+// New creates a new composite runtime host.
+func New(cfg host.Config, provisioner host.Provisioner) (host.Runtime, error) {
+	// Collect available components.
+	availableComps := cfg.Bundle.Manifest.GetAvailableComponents()
+
+	// Collect components that we want.
+	wantedComponents := make(map[bundle.ComponentKind]struct{})
```

Comment:
Hm, the way it currently works is not like that because the node requests components based on type but runtimes may not have the ROFL component for example. What you suggest would make sense if one explicitly configured the required components for each runtime, but I don't think we want to burden the node operators with this?

The only currently required component is RONL.